erdos/op.py

```diff
@@ -86,14 +92,22 @@ def spin(self):
     def log_event(self, processing_time, timestamp, log_message=None):
         pass
 
+    def record_streams(self, stream_uid, msg):
+        pickle.dump(msg, self.recorders[stream_uid])
+
     def _add_input_streams(self, input_streams):
         """Setups and updates all input streams."""
         self.input_streams = self.input_streams + input_streams
+        if self.record_input:
+            for stream in self.input_streams:
```

Comment:
With the change from above, this logic would only set the stream as a recording one.